optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/plugin/rest/model/PortableErrorMessage.java

```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package org.optaweb.vehiclerouting.plugin.websocket;
+package org.optaweb.vehiclerouting.plugin.rest.model;
```

Comment:
OptaWeb Vehicle Routing has a domain-centric architecture inspired by Uncle Bob's [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html) and [Domain-Driven Design](https://airbrake.io/blog/software-design/domain-driven-design). The three top-level packages (`domain`, `service`, `plugin`) are used to organize the code into three layers where source code dependencies are only allowed to go from an outer layer to an inner layer.

Reply:
As much as I understand (and like) this approach, I don't think `plugin` is the right name for this. A plugin is something optional, something extra. To see the word used in these circumstances doesn't feel right.

That said, we certainly do not need to change it right now.

Reply:
I tend to agree. What I like about the term "plugin" is that it's simple and short and it emphasizes that the thing is an _implementation detail_ that can be plugged out or in based on need. In theory that should work. You don't want persistence? Delete the package and provide an alternative repository implementation. But since that is not a functionality that's provided out of the box I understand "plugin" might be confusing.

I don't say we need this but maybe the right way to go about enabling functionality on demand would be Quarkus extensions?

Back to the naming question - any proposals for the package name? I could go with "infrastructure".

Reply:
Honestly, I am not sure the package needs a name, or that it needs to exist. Its purpose is to say "everything here can be replaced by something else". If you want that, it's best said with code - make it actually optional and interchangeable.

That said, I do not see the need for that in the context of a demo application.